tests/unit/amo/components/TestAddon.js

```diff
@@ -404,6 +406,24 @@ describe(__filename, () => {
     expect(root.find('.Addon-title').render().find('a')).toHaveLength(0);
   });
 
+  it('dispatches a server redirect when slug is a numeric ID', () => {
+    const clientApp = CLIENT_APP_FIREFOX;
+    const { store } = dispatchClientMetadata({ clientApp });
+    const addon = createInternalAddon(fakeAddon);
+    store.dispatch(_loadAddons({ addon }));
+
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+    renderComponent({
+      // We set the numeric `id` as slug.
+      params: { slug: addon.id }, store,
+    });
+
+    sinon.assert.calledWith(fakeDispatch, sendServerRedirect({
+      status: 301,
+      url: `/en-US/${clientApp}/addon/${addon.slug}/`,
+    }));
```

Comment:
This could benefit from `sinon.assert.calledOnce()` to make sure it was the only thing dispatched.